Smash ic tau0

zpauliny opened the pull request:
A set of small changes is present in this PR
- change in how tau0 is set when smash IC is used
- several checks and default values for parameters from config file/command line
- option of an extended freezeout.dat write-out that includes energy density and baryon number density

yukarpenko commented:
hi Zuzana - I went through and realized that in the new block in readCommandLine():
   if(strcmp(argv[iarg],"-outputDir")==0) {
    outputDir = argv[iarg+1];}
   else {
    outputDir = "data";
   }
it is assumed that the argument "-outputDir" is always the last one (if I am not mistaken). In case if "-outputDir" is not the last one, the corresponding string variable outputDir will be set to 'data' when the last command-line argument is interpreted, regardless whether -outputDir was specified before. So the -outputDir argument will be ignored if the user puts -outputDir not at the end.
I would suggest to instead default-initialize the outputDir string with "data" when it is declared (line 56; just like you did with other parameters), and remove the else part from the if case I cited.
While we are at it, would it make sense to add a warning like "SMASH IS: tau0 parameter from the input file is ignored" to be printed in case of SMASH IS?

zpauliny commented:
outputDir: yes, I did not realize that, my mistake. I fixed it and also added the warning, good idea! I included a simple header with colour definitions for the console so that the warning stands out (otherwise I did not see it at all). Can be used for warnings and errors in the future. I pushed the changes here.

yukarpenko commented:
Looks good - many thanks for your contribution!
